CMakeLists.txt

```diff
@@ -3,19 +3,35 @@
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 # cmake-format: on
 
-cmake_minimum_required(VERSION 3.10)
+cmake_minimum_required(VERSION 3.27)
 
 project(
   beman_optional26
   VERSION 0.0.0
   LANGUAGES CXX)
 
-enable_testing()
+cmake_policy(VERSION 3.27)
 
 set(TARGETS_EXPORT_NAME ${CMAKE_PROJECT_NAME}Targets)
 
-add_subdirectory(extern)
-add_subdirectory(src)
+# Include the CTest module.
+include(CTest)
+
+# Build the tests only if enabled via the CLI flag: BUILD_TESTING.
+if(BUILD_TESTING)
+  enable_testing()
+
+  # Fetch GoogleTest
+  include(FetchContent)
+  FetchContent_Declare(
```

Comment:
I think without the EXCLUDE_FROM_ALL this is going to build and run the tests from googletest. 
But it also looks like that's not fixed until cmake 3.28.

Reply:
yes, it works.